### PR TITLE
Improve llm response for questions

### LINKED ISSUE #788: [BUG]: Even after setting the requuires sponsorship to YES, the bot automatically selects no radio button
### Describe the bug

Even after setting requires sponsorship to YES and even after giving an example in strings to make the bot answer YES, it still selects NO as the answer.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Branch

None

### Branch name

_No response_

### Python version

_No response_

### LLM Used

_No response_

### Model used

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 49Simon**
Fixes #788 #745 #532 #388
- Pass job application profile info when answering questions
- This will allow the LLM to have all the information required
- Previously, only resume section was provided and it made mistakes answering about legal authorization, self identification questions

### CODE REVIEW

File: src/ai_hawk/llm/llm_manager.py [hunk 1]
```diff
@@ -594,7 +594,7 @@ def answer_question_from_options(self, question: str, options: list[str]) -> str
         prompt = ChatPromptTemplate.from_template(func_template)
         chain = prompt | self.llm_cheap | StrOutputParser()
         output_str = chain.invoke(
-            {"resume": self.resume, "question": question, "options": options})
+            {"resume": self.resume, "job_application_profile": self.job_application_profile, "question": question, "options": options})
```
Comment (surapuramakhil) [inline src/ai_hawk/llm/llm_manager.py:597]: can you move all strings used to constants.py and use them here.

Comment (surapuramakhil) [reply src/ai_hawk/llm/llm_manager.py:597]: Also more selection prompt to prompts.py file. line 438

**Review by surapuramakhil — CHANGES_REQUESTED** [review]

**Comment by surapuramakhil** [comment]
@49Simon can you also think of way to test theses? we would be needing regression for AI responses quality

File: src/ai_hawk/llm/llm_manager.py (hunk 1 above)
Comment (49Simon) [reply src/ai_hawk/llm/llm_manager.py]: I've moved prompts and strings to respective files.

**Review by surapuramakhil — COMMENTED** [review]
There are many more constants being used as keys. try to fix as many as you can.

File: src/ai_hawk/llm/llm_manager.py [hunk 2]
```diff
@@ -71,49 +72,65 @@ def invoke(self, prompt: str) -> BaseMessage:
...
 
+
 class AIAdapter:
     def __init__(self, config: dict, api_key: str):
         self.model = self._create_model(config, api_key)
 
     def _create_model(self, config: dict, api_key: str) -> AIModel:
-        llm_model_type = config['llm_model_type']
-        llm_model = config['llm_model']
+        llm_model_type = config["llm_model_type"]
```
[the remark this answers is not in the document]
Comment (49Simon) [reply src/ai_hawk/llm/llm_manager.py]: moved strings to constant.py

File: src/ai_hawk/llm/llm_manager.py [hunk 3]
```diff
@@ -189,17 +206,19 @@ def log_request(prompts, parsed_reply: Dict[str, Dict]):
...
+            logger.debug(
+                f"Token usage - Input: {input_tokens}, Output: {output_tokens}, Total: {total_tokens}"
+            )
         except KeyError as e:
             logger.error(f"KeyError in parsed_reply structure: {str(e)}")
             raise
 
         try:
-            model_name = parsed_reply["response_metadata"]["model_name"]
+            model_name = parsed_reply[constants.RESPONSE_METADATA][constants.MODEL_NAME]
```
[the remark this answers is not in the document]
Comment (49Simon) [reply src/ai_hawk/llm/llm_manager.py]: changed import statement

**Review by surapuramakhil — APPROVED** [review]

**Pull request merged**